### ISSUE TITLE
Quoting "class" & "for" attributes in compiled JSX

### ISSUE THREAD

**Issue by jcfranco** [Microsoft/TypeScript#9827]
This came up from a project using a custom JSX factory (`--reactNamespace`). The custom factory does not support `classNames` and instead relies on `class`. In this workflow, `class` isn't quoted when compiled (the same would happen with `for`). The goal would be to consume the output source from the TypeScript compiler without any post-processing. Could TypeScript be updated to quote these attributes when compiling?

**Example**

With the following render method:

```jsx
render() {
  return (
    <div>
      <label class="label-class" for="my-input"></label>
      <input id="my-input"/>
    </div>
  );
}
```

**Expected output**

```js
Test.prototype.render = function () {
    return (/* ... */, customFactory.createElement("label", {"class": "label-class", "for": "my-input"}), /* ... */);
};
```

**Actual output** 

```js
Test.prototype.render = function () {
    return (/* ... */, customFactory.createElement("label", {class: "label-class", for: "my-input"}), /* ... */);
};
```

**Comment by danpantry** [comment]
I'd be inclined to argue that `class` and `for` should be warnings at compile time when used with React (as they are at runtime). You're *meant* to use `className` and `htmlFor` as `class` and `for` are keywords.

**Comment by RyanCavanaugh** [comment]
I'm a little confused; why should we bother quoting these names? This isn't a syntax error.

**Comment by jcfranco** [comment]
@RyanCavanaugh This came up because I kept getting JSHint errors from the compiled JS. I originally thought this error was triggered by the reserved word, but after your previous comment I dug a little deeper and the culprit was an outdated JSHint configuration file with `es5: false`. #facepalm

Sorry for the hassle. 

@danpantry Interesting suggestion. Probably an issue for React since the runtime warning is not logged by TypeScript.

**Comment by danpantry** [comment]
@jcfranco I meant I would expect the `.tsx`/`.jsx` compiler to emit a warning in TS rather than wait until runtime :)

**Comment by jcfranco** [comment]
@danpantry Ah, gotcha.

**Comment by danpantry** [comment]
holy quick reply batman
